tubelib/node_states.lua

```diff
@@ -175,9 +176,27 @@ function NodeStates:start(pos, meta, called_from_on_timer)
 		if self.formspec_func then
 			meta:set_string("formspec", self.formspec_func(self, pos, meta))
 		end
-		minetest.get_node_timer(pos):start(self.cycle_time)
+		local cycle_time = self.cycle_time
+		if self.first_cycle_time then
+			if meta:get_int("tubelib_first_run") == 1 then
+				meta:set_int("tubelib_first_run", 0)
+				cycle_time = self.cycle_time
+			else
+				meta:set_int("tubelib_first_run", 1)
+				cycle_time = self.first_cycle_time
+			end
+		end
+		minetest.get_node_timer(pos):start(cycle_time)
 		return true
 	end
+	if self.first_cycle_time and meta:get_int("tubelib_first_run") == 1 then
+		local cycle_time = self.cycle_time
+		local timer = minetest.get_node_timer(pos)
+		minetest.after(0, function ()
+			timer:set(cycle_time, timer:get_elapsed())
+		end)
```

Comment:
I really feel like this `minetest.after` should not be required. Perhaps it's because it will not let us modify a timer inside the callback for the timer

Reply:
Yes, you can't restart the timer from the timer callback